### DISCUSSION TITLE
Month and Year picker

### DISCUSSION THREAD

**Discussion by Berdugo1994** [heroui-inc/heroui#3630] [Feature requests]
Using the awesome DatePicker component, I don't see an option to select a month & year tuple as a value.
Making granularity="month" option valid could be even more awesome than it now. Thanks in advance :)

**Comment by elRatto21** [comment]
I agree, I need this too

**Comment by pmt1506** [comment]
i also need this too, hope they include it

**Comment by AnshDubey1999** [comment]
This would be really nice, ++1

**Comment by ekajaya740** [comment]
+1

Its would be nice if this implemented

But if you need it right now, I've found an alternative way that fits my needs to display only month and year by applying this `props` to the `DatePicker`

```js
showMonthAndYearPickers
calendarProps={{
    isHeaderExpanded: true,
    hideDisabledDates: true,
}}
```

**Reply by fahry-mohammed** [reply]
Thanks, UI is okay which shows years and months. But selecting year and month, DatePicker modal not getting close nor showing selected values. How did you handle that?

**Reply by ekajaya740** [reply]
Hi @fahry-mohammed 

I just set the date to 1 and restrict the date to only select 1. Its working without problem for me.

```js

isDateUnavailable={ (date) =>
	!dayjs(date).isValid() || dayjs(date).date() !== 1
}
```

But now I'm still figuring out on how to hide the date, so it will displayed in `MM/YYYY` format.

**Reply by fahry-mohammed** [reply]
@ekajaya740, Btw I ended up creating a new component called `YearMonthPicker` our own using NextUI’s Popover!

**Reply by ekajaya740** [reply]
@fahry-mohammed Cool, I'm curious on how you do it. If you don't mind, I would like to see your implementation.

**Reply by fahry-mohammed** [reply]
@ekajaya740 Here you go: [CvManagerMonthYearPicker](https://gist.github.com/fahry-mohammed/2fbf4e840ce53518388364cc532e5683). Please let me know if you find any bugs or improvement.

**Comment by lanzclintonv** [comment]
This would be good if we had this component. Also to add in, a quarter picker too. AntDesign has these all.

**Comment by fahry-mohammed** [comment]
I ended up creating a new component called MonthYearPicker our own using NextUI’s Popover: [CvManagerMonthYearPicker](https://gist.github.com/fahry-mohammed/2fbf4e840ce53518388364cc532e5683). Please let me know if you find any bugs or improvement.
